tensornetwork/network.py

```diff
@@ -913,6 +914,32 @@ def split_node_full_svd(
     self.nodes_set.remove(node)
     return left_node, singular_values_node, right_node, trun_vals
 
+  def remove_node(self, node: network_components.Node
+    ) -> Dict[Union[int, Text], network_components.Edge]:
+    """Remove a node from the network.
+
+    Args:
+      node: The node to be removed.
+
+    Returns:
+      A dictonary mapping name/index to the new broken edge.
+
+    Raises:
+      ValueError: If the node isn't in the network.
+    """
+    if node not in self:
+      raise ValueError("Node '{}' is not in the network.".format(node))
+    broken_edges = {}
+    for i, name in enumerate(node.axis_names):
+      if not node[i].is_dangling() and not node[i].is_trace():
```

Comment:
After `remove_node()` the dangling and trace edges of the node are ripe for garbage collection. However, by leaving them in the state where they refer to each other we create reference cycles that aren't handled by the reference counting mechanism.

I think the garbage collector may still handle those, but it's easy and probably safer to just disconnect the dangling and trace edges from the node here. WDYT?

Reply:
So, since edges have weak reference to their nodes, the only strong references are Node -> Edge, so once the Node is garbage collected, the edges will be too.

I don't want to fully disconnect these edges, as it is conceivable that someone might want to reconnect that node to another part of the network.